en,zh: add v1.5.4 release notes (#2620)
pingcap/docs-tidb-operator: en/releases/release-1.5.4.md

---
title: TiDB Operator 1.5.4 Release Notes
summary: Learn about new features and bug fixes in TiDB Operator 1.5.4.
---

# TiDB Operator 1.5.4 Release Notes

Release date: September 13, 2024

TiDB Operator version: 1.5.4

## Improvements

- VolumeReplace feature supports customizing the number of spare replicas used during TiKV or PD disk replacements ([#5666](https://github.com/pingcap/tidb-operator/pull/5666), [@anish-db](https://github.com/anish-db))
- VolumeReplace feature supports only enabling for specified TiDB clusters ([#5670](https://github.com/pingcap/tidb-operator/pull/5670), [@rajsuvariya](https://github.com/rajsuvariya))

> 

Reviewer: - EBS Snapshot restore supports to configure if warmup failure on individual volumes fails the restore instantly ([#5635](https://github.com/pingcap/tidb-operator/pull/5635), [@michealdeng](https://github.com/[michaelmdeng](https://github.com/michaelmdeng)))
- Gate restore failure on warmup failure only for `check-wal-only` warmup strategy. ([#5636](https://github.com/pingcap/tidb-operator/pull/5636), [@michealdeng](https://github.com/[michaelmdeng](https://github.com/michaelmdeng)))
- support to fetch backup size from backupmeta ([#5430](https://github.com/pingcap/tidb-operator/pull/5430), [@Leavrth](https://github.com/Leavrth))

## Bug fixes

> - Fix the issue that tidb-backup-manager can't parse files storage size from BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))

Reviewer: ```suggestion
- Fix the issue that tidb-backup-manager can't parse files storage size from BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))
- cleanup EBS volumes if EBS snapshot restore fails ([#5639](https://github.com/pingcap/tidb-operator/pull/5639), [@WangLe1321](https://github.com/WangLe1321))
- Reset metrics when federated manager restarts ([#5648](https://github.com/pingcap/tidb-operator/pull/5648), [@wxiaomou](https://github.com/wxiaomou))
- Enhance TC check for EBS Snapshot and fail EBS snapshot backup if no TiKV nodes or replica is 0 ([#5662](https://github.com/pingcap/tidb-operator/pull/5662), [@BornChanger ](https://github.com/BornChanger))
```

---
title: TiDB Operator 1.5.4 Release Notes

> summary: Learn about new features and bug fixes in TiDB Operator 1.5.4.

Reviewer 2: ```suggestion
summary: Learn about improvements and bug fixes in TiDB Operator 1.5.4.
```

---

# TiDB Operator 1.5.4 Release Notes

Release date: September 13, 2024

TiDB Operator version: 1.5.4

## Improvements

> - VolumeReplace feature supports customizing the number of spare replicas used during TiKV or PD disk replacements ([#5666](https://github.com/pingcap/tidb-operator/pull/5666), [@anish-db](https://github.com/anish-db))
> - VolumeReplace feature supports only enabling for specified TiDB clusters ([#5670](https://github.com/pingcap/tidb-operator/pull/5670), [@rajsuvariya](https://github.com/rajsuvariya))
> - EBS Snapshot restore supports to configure if warmup failure on individual volume fails the restore instantly ([#5622](https://github.com/pingcap/tidb-operator/pull/5622), [@michaelmdeng](https://github.com/michaelmdeng))
> - Gate EBS Snapshot restore failure on warmup failure only for `check-wal-only` warmup strategy ([#5621](https://github.com/pingcap/tidb-operator/pull/5621), [@michaelmdeng](https://github.com/michaelmdeng))

Reviewer 3: ```suggestion
- The VolumeReplace feature supports customizing the number of spare replicas for PD and TiKV ([#5666](https://github.com/pingcap/tidb-operator/pull/5666), [@anish-db](https://github.com/anish-db))
- The VolumeReplace feature can be enabled for specific TiDB clusters ([#5670](https://github.com/pingcap/tidb-operator/pull/5670), [@rajsuvariya](https://github.com/rajsuvariya))
- EBS snapshot restore supports configuring whether to terminate the entire restore task immediately if volume warmup fails ([#5622](https://github.com/pingcap/tidb-operator/pull/5622), [@michaelmdeng](https://github.com/michaelmdeng))
- When using the `check-wal-only` warmup strategy, EBS snapshot restore marks the entire restore task as failed if warmup fails ([#5621](https://github.com/pingcap/tidb-operator/pull/5621), [@michaelmdeng](https://github.com/michaelmdeng))
```

## Bug fixes

> - Fix the issue that tidb-backup-manager cannot parse backup file storage size in BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))

Reviewer 2: ```suggestion
- Fix the issue that `tidb-backup-manager` cannot parse backup file size in BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))
```

> - Fix a potential EBS volume leak issue when EBS snapshot restore fails ([#5634](https://github.com/pingcap/tidb-operator/pull/5634), [@WangLe1321](https://github.com/WangLe1321))

Reviewer 2: ```suggestion
- Fix a potential EBS volume leak issue that occurs when EBS snapshot restore fails ([#5634](https://github.com/pingcap/tidb-operator/pull/5634), [@WangLe1321](https://github.com/WangLe1321))
```

> - Fix the issue that tidb-backup-manager can't parse files storage size from BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))
> - Fix the issue that EBS may leak after EBS snapshot restore fails ([#5634](https://github.com/pingcap/tidb-operator/pull/5634), [@WangLe1321](https://github.com/WangLe1321))
> - Fix the issue that the metrics are not initialized after the federated manager restarts ([#5637](https://github.com/pingcap/tidb-operator/pull/5637), [@wxiaomou](https://github.com/wxiaomou))
> - Fix the issue that EBS snapshot restore doesn't fail even if no TiKV instance configured or replica is set to 0 ([#5659](https://github.com/pingcap/tidb-operator/pull/5659), [@BornChanger](https://github.com/BornChanger))

Reviewer 3: ```suggestion
- Fix the issue that tidb-backup-manager cannot parse backup file storage size in BR backupmeta v2 ([#5411](https://github.com/pingcap/tidb-operator/pull/5411), [@Leavrth](https://github.com/Leavrth))
- Fix a potential EBS volume leak issue when EBS snapshot restore fails ([#5634](https://github.com/pingcap/tidb-operator/pull/5634), [@WangLe1321](https://github.com/WangLe1321))
- Fix the issue that metrics are not properly initialized after federated manager restarts ([#5637](https://github.com/pingcap/tidb-operator/pull/5637), [@wxiaomou](https://github.com/wxiaomou))
- Fix the issue that EBS snapshot restore does not fail when no TiKV instances are configured or TiKV replica is set to 0 ([#5659](https://github.com/pingcap/tidb-operator/pull/5659), [@BornChanger](https://github.com/BornChanger))
```

> - Fix the issue that EBS snapshot restore does not fail when no TiKV instances are configured or TiKV replica is set to 0 ([#5659](https://github.com/pingcap/tidb-operator/pull/5659), [@BornChanger](https://github.com/BornChanger))

Reviewer 3: ```suggestion
- Fix the issue that EBS snapshot restore incorrectly succeeds when no TiKV instances are configured or TiKV replica is set to 0 ([#5659](https://github.com/pingcap/tidb-operator/pull/5659), [@BornChanger](https://github.com/BornChanger))
```

Reviewer 3: @csuzhangxc PTAL agin
